google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddResponsiveSearchAdWithAdCustomizer.java

```diff
@@ -0,0 +1,261 @@
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.adGroupId = Long.parseLong("INSERT_AD_GROUP_ID_HERE");
+
+      // Optional: Specify a customizerAttributeName. The default value in this example is "Price".
+      params.customizerAttributeName = "Price";
+    }
+
+    GoogleAdsClient googleAdsClient = null;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      System.exit(1);
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      System.exit(1);
+    }
+
+    try {
+      new AddResponsiveSearchAdWithAdCustomizer()
+          .runExample(
+              googleAdsClient, params.customerId, params.adGroupId, params.customizerAttributeName);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupId the ad group ID.
+   * @param customizerAttributeName the customizer attribute name.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      long adGroupId,
+      String customizerAttributeName) {
+    String customizerAttributeResourceName =
+        createCustomizerAttribute(googleAdsClient, customerId, customizerAttributeName);
+    linkCustomizerAttributeToCustomer(googleAdsClient, customerId, customizerAttributeResourceName);
+    createResponsiveSearchAdWithCustomization(
+        googleAdsClient, customerId, adGroupId, customizerAttributeName);
+  }
+
+  // [START add_responsive_search_ad_with_ad_customizer_1]
+  /** Creates a customizer attribute with the specified customizer attribute name. */
+  private static String createCustomizerAttribute(
+      GoogleAdsClient googleAdsClient, long customerId, String customizerAttributeName) {
+    // Creates a customizer attribute with the specified name.
+    CustomizerAttribute customizerAttribute =
+        CustomizerAttribute.newBuilder()
+            .setName(customizerAttributeName)
+            .setType(CustomizerAttributeType.PRICE)
```

Comment:
Nit: Add why we use price here:

"
// Specifies the type to be 'PRICE' so that we can dynamically customize the part of the ad's description that is a price of a product/service we advertise.
"

Reply:
Fixed.